syncer/main.py

```diff
@@ -54,137 +127,369 @@ def __init__(self, webinterface, connector):
+            username = gateway.get('username', '')
+            password = gateway.get('password', '')
+
+            if gateway.get('remote_name', '') == '':
+                gw_name = ip
+            else:
+                gw_name = gateway.get('remote_name')
+
+            headers = {
+                'X-Requested-With': 'OpenMotics plugin: Syncer'
+            }
+            endpoint = 'https://{0}'.format(ip)
 
-        self._sensor_mapping = {}
-        for entry in self._config.get('sensors', []):
+            enabled = ip != '' and username != '' and password != ''
+            # If one GW is enabled, then the plugin will be enabled as well. But this is done after gathering all the
+            # config of the other GWs.
+            if enabled:
+                _enabled = True
+            self._gateways[ip] = {
+                'ip': ip,
+                'name': gw_name,
+                'username': username,
+                'password': password,
+                'headers': headers,
+                'endpoint': endpoint,
+                'enabled': enabled
+            }
             try:
-                self._sensor_mapping[int(entry['local_sensor_id'])] = int(entry['remote_sensor_id'])
+                if len(gateway.get('sensors', [])) > 0:
+                    remote_sensor_conf = self._call_remote('get_sensor_configurations', self._gateways[ip]).get(
+                        'config')
+                else:
+                    remote_sensor_conf = []
+                if len(gateway.get('outputs', [])) > 0:
+                    remote_output_conf = self._call_remote('get_output_configurations', self._gateways[ip]).get(
+                        'config')
+                else:
+                    remote_output_conf = []
+                if len(gateway.get('shutters', [])) > 0:
+                    remote_shutter_conf = self._call_remote('get_shutter_configurations', self._gateways[ip]).get(
+                        'config')
+                else:
+                    remote_shutter_conf = []
             except Exception as ex:
-                logger.exception('Could not load temperature mapping: {0}'.format(ex))
+                raise RuntimeError(f"Couldn't connect to GW {ip}: {ex}")
 
-        self._output_mapping = {}
-        for entry in self._config.get('outputs', []):
+            # Specific for each remote GW as this is remote 2 local. A loop over every remote GW will occur and this
+            # will give our local sensor_dto if a value has changed.
+            sensor_mapping = {}
+            for entry in gateway.get('sensors', []):
+                try:
+                    remote_sensor_id = entry.get('remote_sensor_id', -1)
+                    if remote_sensor_id < 0:
+                        raise RuntimeError("Please give valid sensor ids")
+                    external_id = f"syncer/{ip}/{remote_sensor_id}"
+                    for remote_sensor in remote_sensor_conf:
```

Comment:
instead of looping over all the sensors in the remote_sensor_conf, populate them in a dict with the key the id of the remote sensor. Then your lookup will be something like
```
remote_sensor_conf.get(remote_sensor_id, None)
```

Reply:
Exactly what I did with the sensor_mapping variable => {remote_sensor_id: local_sensor_tdo}